### PR TITLE
Fix and Optimize CI

### PR DESCRIPTION
**Pull request by maxsxu**
Fixes based on #950

### Motivation

Fix and optimize CI as the TLS certificates for CI are expired and many unnecessary CI trigger causes slow CI.

### Modifications

- Update TLS certificates and add scripts to generate certs
- Reduce unnecessary CI trigger for sn-platform chart
- Consolidate CI workflow jobs for pulsar chart
- Optimize scripts

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is already covered by existing tests.

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by zymap — APPROVED** [review]
LGTM

File: .ci/chart_test.sh [hunk 1]
```diff
@@ -19,7 +19,9 @@ source ${CHARTS_HOME}/.ci/helm.sh
 ci::create_cluster
 
 # delete the cluster
-trap 'ci::delete_cluster' 0
+# TODO: Make cluster cleanup as a sepratee steep.
```
Comment (ciiiii) [inline .ci/chart_test.sh:22]: small typo

**Review by ciiiii — DISMISSED** [review]
LGTM
Another question maybe we can improve in the future: can we generate these tls certs with cert-manager in ci state?

File: .ci/chart_test.sh (hunk 1 above)
Comment (ericsyh) [reply .ci/chart_test.sh:22]: @maxsxu pls update this typo then we will merge this PR.

**Review by ciiiii — APPROVED** [review]

File: .ci/chart_test.sh [hunk 2]
```diff
@@ -19,7 +19,9 @@ source ${CHARTS_HOME}/.ci/helm.sh
 ci::create_cluster
 
 # delete the cluster
-trap 'ci::delete_cluster' 0
+# TODO: Make cluster cleanup as a separate steep.
```
Comment (ericsyh) [inline .ci/chart_test.sh:22]:
```suggestion
# TODO: Make cluster cleanup as a separate step.
```

**Review by ericsyh — APPROVED** [review]

**Review by ericsyh — APPROVED** [review]

**Pull request merged**